src/groups/group.jl

```diff
@@ -45,7 +68,7 @@ Group manifolds by default forward metric-related operations to the wrapped mani
     GroupManifold(manifold, op)
 """
 struct GroupManifold{𝔽,M<:AbstractManifold{𝔽},O<:AbstractGroupOperation} <:
```

Comment:
I'd make the transparency type a parameter of `GroupManifold`. And it looks weird that `DefaultGroupDecoratorType` is not the default, probably the name is misleading.

Reply:
That would mean quite some rework, but sure we can do that. The naming follows that of the `EmbeddedManifold`, and the transparent case is used far more often. With a Parameter in `GroupManifold` there is no default anymore, but we have to add the transparent type then for all but 2 groups I think.

Reply:
I gave it a try and made it a parameter of the group manifold. That yields about 100+ errors so possibly _quite_ _quite_ a lot of rework and much breaking changes.

Reply:
For example the constructor will not work anymore and we would have to think about how to model that. That constructor is heavily used, too.

Reply:
Currently `GroupManifold` is used to define `SemidirectProductGroup`, and that is used for `SpecialEuclidean`. And `SpecialEuclidean` needs to have different connections available, and only one of them would be transparent. Similarly for `SpecialOrthogonal`. This isn't needed for the current state of Manifolds.jl but I think I need this for #381 . Anyway, I can also add this parameter in my PR :slightly_smiling_face: .

Reply:
OK, so I'll do this rework in #381.

Reply:
Ok, great if you could take that over. I tried and for me it seems like at least half a day to rework this main type of group manifolds. And I did this rework already when introducing the transparency type here. It is not even a fun work.